status_and_problems.tex

```diff
@@ -0,0 +1,132 @@
+Just as an example, it is used to create the same tag on all the GitHub packages composing a defined product automatically, instead of creating it manually.
+
+\textit{TO BE CLARIFIED: which are java build tools?}
+
+
+\subsection{Open Problems} \label{sec:openProblems}
+
+In this section a list of problems derived from the current development approach are enumerated. 
+
+While these problems are unresolved the only release process suitable for DM is to release the entire codebase each time.
+
+Their resolution may lead to a different development approach, and therefore the proposed release procedure may need to be reviewed accordingly.
+
+
+\subsubsection{SW Product Composition} \label{sec:problemId}
+
+A GitHub package may be included in more than one software product.
+
+This make impossible to apply in a consistent way the release procedure to a SW products that is not the Science Pipelines.
+The reason is explained in the following example.
+
+Product A is composed by 100 Git packages.
+
+Product B is composed by 80 Git packages.
+
+70 Git packages are shared by both products.
+
+When release 1.0 is done on product A, all 100 packages will be tagged, and corresponding Eups packages are created. This will include the 70 shared packages.
+
+One week lather, product B is ready for the release 1.0, and some of the 70 shared packages have been updated.
+For them, the 1.0 tag required for product B release, will not be the same as the 1.0 tag required for the product A release.
+This implies that for some packages, release 1.0 of product A, is different from release 1.0 of product B.
+
+\textbf{Requirement}: there shall be a 1 to 1 correspondence between software products and GitHub packaged. If this is not possible, each GitHub package shall be included in only one software product.
+Note also that, each GitHub package used for building, unit testing and packaging the software products shall not be included in the SW product itself, but versioned separately and be part of the environment definition. See section \ref{sec:swprod}.
+
+
+\subsubsection{Code Fragmentation} \label{sec:problemCode}
+
+The high number of packages causes problems in that it :
+
+\begin{itemize}
+\item increases the build time: each Git package needs to be build each time.
+\item increases the release time: all Git packages need to be released each time.
+\item increases the failure probability: the tooling may be affected by network glitch or similar technical issues. This may lead to the failure of the build/release process in a non deterministic way.
+\end{itemize}
+
+Moving 3rd party libraries to conda environment, will reduce this problem. 
+However this requires proper management of the conda environment.
+
+\textbf{Requirement}: the number of Git packages shall be kept low. DM-CCB shall approve each time a new package is introduced, since this has an impact on the maintainability of the system. 
```

Comment:
Can we quantify “low”? I can't tell if this section is simply suggesting that we should avoid adding new packages in future, or if it's advocating that existing packages be merged or eliminated.

Reply:
It is difficult to quantify. It depends on the use case. If you want to have the Alert Product software available quickly in case of fix, you cant have 120 git packages to release, because this will take 12 hours minimum, and you may want to have the fix available in 30 minutes... get this as an example. Such a problem may not exist for DRP software instead.